src/Csv.php

```diff
@@ -540,6 +540,41 @@ public function auto($file = null, $parse = true, $search_depth = null, $preferr
         return $this->delimiter;
     }
 
+    /**
+     * Get total number of rows in csv without parsing whole data.
+     *
+     * @return bool|int
+     */
+    public function getTotalRowCount(){
+        if (empty($this->file_data)){
+            return false;
+        }
+
+        $this->_detect_and_remove_sep_row_from_data($this->file_data);
+
+        $pattern = sprintf('/("[^%s]*")|[^%s]*/i',$this->enclosure, $this->enclosure);
+        preg_match_all($pattern,$this->file_data, $matches);
+
+        foreach ($matches[0] as $match){
+            if (empty($match) || !preg_match("/{$this->enclosure}/", $match)){
+                continue;
+            }
+
+            $replace = str_replace(["\r", "\n"], '', $match);
+            $this->file_data = str_replace($match, $replace, $this->file_data);
```

Comment:
Is there a benefit of "messing" with file_data? Without knowing, I'd suspect a local string variable is better suited here.

Reply:
Yeah, local variable is correct in this situation

Reply:
Done